ring_doorbell/other.py

```diff
@@ -0,0 +1,287 @@
+        self._ring.update_devices()
+        return True
+
+    @property
+    def keep_alive_auto(self):
+        if self.kind in INTERCOM_KINDS:
+            return self._attrs.get("settings").get("keep_alive_auto")
+        return None
+
+    @keep_alive_auto.setter
+    def keep_alive_auto(self, value):
+        url = SETTINGS_ENDPOINT.format(self.device_api_id)
+        payload = {"keep_alive_settings": {"keep_alive_auto": value}}
+
+        self._ring.query(url, method="PATCH", json=payload)
+        self._ring.update_devices()
+        return True
+
+    @property
+    def mic_volume(self):
+        """Return mic volume."""
+        if self.kind in INTERCOM_KINDS:
+            return self._attrs.get("settings").get("mic_volume")
+        return None
+
+    @mic_volume.setter
+    def mic_volume(self, value):
+        if not ((isinstance(value, int)) and (MIC_VOL_MIN <= value <= MIC_VOL_MAX)):
+            _LOGGER.error("%s", MSG_VOL_OUTBOUND.format(MIC_VOL_MIN, MIC_VOL_MAX))
+            return False
+
+        url = SETTINGS_ENDPOINT.format(self.device_api_id)
+        payload = {"volume_settings": {"mic_volume": value}}
+
+        self._ring.query(url, method="PATCH", json=payload)
+        self._ring.update_devices()
+        return True
+
+    @property
+    def voice_volume(self):
+        """Return voice volume."""
+        if self.kind in INTERCOM_KINDS:
+            return self._attrs.get("settings").get("voice_volume")
+        return None
+
+    @voice_volume.setter
+    def voice_volume(self, value):
+        if not ((isinstance(value, int)) and (VOICE_VOL_MIN <= value <= VOICE_VOL_MAX)):
+            _LOGGER.error("%s", MSG_VOL_OUTBOUND.format(VOICE_VOL_MIN, VOICE_VOL_MAX))
+            return False
+
+        url = SETTINGS_ENDPOINT.format(self.device_api_id)
+        payload = {"volume_settings": {"voice_volume": value}}
+
+        self._ring.query(url, method="PATCH", json=payload)
+        self._ring.update_devices()
+        return True
+
+    @property
+    def clip_length_max(self):
+        # this value sets an effective refractory period on consecutive rigns
+        # eg if set to default value of 60, rings occuring with 60 seconds of
+        # first will not be detected
+
+        url = SETTINGS_ENDPOINT.format(self.device_api_id)
+
+        return (
+            self._ring.query(url, method="GET")
+            .json()
+            .get("video_settings")
+            .get("clip_length_max")
+        )
+
+    @clip_length_max.setter
+    def clip_length_max(self, value):
+        url = SETTINGS_ENDPOINT.format(self.device_api_id)
+        payload = {"video_settings": {"clip_length_max": value}}
+        try:
+            self._ring.query(url, method="PATCH", json=payload)
+            self._ring.update_devices()
+            return True
+        except Exception as ex:
+            raise RingError(f"Unknown error during query of url {url}: {ex}") from ex
```

Comment:
Why is this catch needed as `query` should only raise `RingError`.  Was it for debugging?

https://github.com/tchellomello/python-ring-doorbell/blob/master/ring_doorbell/auth.py#L155